### ISSUE TITLE
ConcurrentModificationException in HashMap

### ISSUE THREAD

**Issue by philippb** [amplitude/Amplitude-Android#9]
```
java.util.ConcurrentModificationException
at java.util.HashMap$HashIterator.nextEntry(HashMap.java:796)
at java.util.HashMap$EntryIterator.next(HashMap.java:833)
at java.util.HashMap$EntryIterator.next(HashMap.java:831)
at org.json.JSONObject.writeTo(JSONObject.java:666)
at org.json.JSONStringer.value(JSONStringer.java:237)
at org.json.JSONObject.writeTo(JSONObject.java:667)
at org.json.JSONObject.toString(JSONObject.java:636)
at com.amplitude.api.Amplitude.logEvent(Unknown Source)
at com.amplitude.api.Amplitude.logEvent(Unknown Source)
at com.amplitude.api.Amplitude.startNewSession(Unknown Source)
at com.amplitude.api.Amplitude.startNewSessionIfNeeded(Unknown Source)
at com.amplitude.api.Amplitude.access$600(Unknown Source)
at com.amplitude.api.Amplitude$4.run(Unknown Source)
at android.os.Handler.handleCallback(Handler.java:587)
at android.os.Handler.dispatchMessage(Handler.java:92)
at android.os.Looper.loop(Looper.java:130)
at android.os.HandlerThread.run(HandlerThread.java:60)
```

**Comment by philippb** [comment]
This is still an issue in version 1.3

![crash distribution](https://photos-2.dropbox.com/t/2/AAB9nQ8cbtz3O1l10p-0wgdXSzGqIVJwDNKPg0EK5lrCLQ/12/620820/png/1024x768/3/1425067200/0/2/Screenshot%202015-02-27%2010.49.51.png/CJTyJSABIAIgAygBKAI/07eubot9Iq4piT6zgHIYgH1IBbXapIxWOUI_AKv4ELQ)

```
0	java.util.ConcurrentModificationException
1		at java.util.HashMap$HashIterator.nextEntry(HashMap.java:792)
2		at java.util.HashMap$EntryIterator.next(HashMap.java:829)
3		at java.util.HashMap$EntryIterator.next(HashMap.java:827)
4		at org.json.JSONObject.writeTo(JSONObject.java:666)
5		at org.json.JSONStringer.value(JSONStringer.java:237)
6		at org.json.JSONObject.writeTo(JSONObject.java:667)
7		at org.json.JSONObject.toString(JSONObject.java:636)
8		at com.amplitude.api.Amplitude.logEvent(Unknown Source)
9	                                logEvent
10	                                logEvent
11	                                logEvent
12		at com.amplitude.api.Amplitude.logEvent(Unknown Source)
13	                                logEvent
14	                                logEvent
15	                                logEvent
16		at com.amplitude.api.Amplitude.access$000(Unknown Source)
17		at com.amplitude.api.Amplitude$1.run(Unknown Source)
18		at android.os.Handler.handleCallback(Handler.java:725)
19		at android.os.Handler.dispatchMessage(Handler.java:92)
20		at android.os.Looper.loop(Looper.java:137)
21		at android.os.HandlerThread.run(HandlerThread.java:60)

```

**Comment by allanca** [comment]
Fixed in release 1.4.4 on 3/11.
